app/main.py

```diff
@@ -1,6 +1,41 @@
 from typing import Callable
+import functools
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_storage = {}  # Dictionary to store results of completed runs
+
+    @functools.wraps(func)
+    def wrapper(*args) -> Callable:
+        if args in cache_storage:
+            print("Getting from cache")
+            return cache_storage[args]
+        else:
+            print("Calculating new result")
+            result = func(*args)
+            cache_storage[args] = result
```

Comment:
The cache_storage dictionary is a mutable object and is being modified here. You do not need to use the nonlocal keyword in this case.